scripts/test_release_artifacts.py

```
@@ -339,36 +373,113 @@ def _executable_lines(script: str) -> list[str]:
-        # Release assets stay mutable while the release is a draft, and the job
-        # holds contents: write, so verifying and then promoting in one step
-        # NARROWS the window in which a verified manifest could be replaced. It
-        # does not close it. Preventing the substitution outright needs an
-        # immutable or pinned publication mechanism, which GitHub releases do not
-        # provide.
-        #
-        # Both commands must live inside the promotion step for even that
-        # narrowing to hold, so the ordering is asserted within the step's own
-        # text. Searching the whole workflow would pass while the two sat in
-        # different steps, which is the arrangement this is meant to rule out.
+    def test_promotion_is_separate_protected_and_signature_gated(self) -> None:
```

Ruff (0.16.1)

[warning] Too many statements (79 > 50)

(PLR0915)